components/eamxx/src/control/intensive_observation_period.cpp

```diff
@@ -2,11 +2,17 @@
 
 #include "share/grid/point_grid.hpp"
 #include "share/io/scorpio_input.hpp"
+#include "share/util/scream_vertical_interpolation.hpp"
 
 #include "ekat/ekat_assert.hpp"
+#include "ekat/util/ekat_lin_interp.hpp"
 
-// Extend ekat mpi type for pair of double int, used
-// in find_closest_lat_lon_index_and_rank()
+#include "pio.h"
```

Comment:
Yes, we need it or else it complains about all the `PIO_*` variables/function needed in `read_dimensionless_variable_from_file()`.

Reply:
I wouldn't mind moving it. Would it make sense adding the `read_variable_from_file()` as well? Maybe they could be combined as a single function (and the user gives an empty set of dimensions for "dimensionless" version?).

Reply:
Yeah, that seems like a fine way to avoid multiple very similar declarations..

The reason for these was that you didn't want to declare new field tags, right? If so, as an alternative, I can probably get rid of these two once I merge the #2620 ... Then, you should be able to use arbitrary names for your tags (e.g., create a layout with tags `<CMP,CMP>` and names `("iop_dim1", "iop_dim2")`). Would that work?

Reply:
Yes, I'm looking through that PR now. That would definitely work. Maybe there are some unknowns with rank0 fields and the `AtmosphereInput` class? But that should be fixed if there are anyways.

Looking at some of the reads I'm doing here, these `read_variable_from_file()` functions definitely simplify the code in some cases. For instance, `bdate`, `lat`, `lon`, `lev` (pressures), and the surface values all don't have fields associated with them, so using `AtmosphereInput` class would add a decent bit of extra code. What do you think of using both? The majority of the iop file data (associated with fields) can be read using `AtmosphereInput` which I think would make the `read_iop_file_data()` function much cleaner, and then the more one-off cases could use the `read_variable_from_file()` function.

Also I'm guessing your PR can get merged earlier, since I'll still need some more eyes on this one, so I don't mind waiting here to incorporate those changes.

Reply:
Yes, it's fine to use scorpio interface functions instead of `AtmosphereInput` if the data is not naturally available as a `Field`.